src/_shared/js/capi-multiple.js

```diff
@@ -0,0 +1,262 @@
+
+	if (OVERRIDES.images[cardNumber] !== '') {
+		backupImg.src = OVERRIDES.images[cardNumber];
+	} else {
+
+		let sources = buildSources(cardInfo.articleImage.sources);
+		insertBetweenComments(card, sources);
+		backupImg.src = cardInfo.articleImage.backupSrc;
+
+	}
+
+}
+
+// Constructs an individual card.
+function buildCard (cardInfo, cardNumber, isPaid) {
+
+	let adType = isPaid ? 'paidfor' : 'supported';
+	let cardTemplate = document.getElementById(`${adType}-card`);
+	let cardFragment = document.importNode(cardTemplate.content, true);
+	let card = cardFragment.querySelector(`.advert--${adType}`);
+
+	buildTitle(card, cardInfo, cardNumber);
+	card.querySelector('a.advert').href = cardInfo.articleUrl;
+	insertImages(card, cardInfo, cardNumber);
+
+	// Only first two cards show on mobile portrait.
+	if (cardNumber >= 2) {
+		card.classList.add('hide-until-mobile-landscape');
+	}
+
+	return cardFragment;
+
+}
+
+// Adds branding information from cAPI or DFP override.
+function addBranding (brandingCard, isPaid) {
+
+	let imageClass = `.${isPaid ? 'adverts__' : ''}badge__logo`;
+	let brandImage = document.querySelector(imageClass);
+
+	if (OVERRIDES.brandLogo === '') {
+		brandImage.src = brandingCard.branding.sponsorLogo.url;
+	}
+
+}
+
+// Sets correct glabs link based on edition (AU/All others).
+function editionLink (edition, isPaid) {
+
+	if (isPaid) {
+		let link = edition === 'AU' ? GLABS_EDITION.au : GLABS_EDITION.default;
+		document.querySelector('.adverts__stamp a').href = link;
+	}
+
+}
+
+// Constructs an array of cards from an array of data.
+function buildCards (cardsInfo, isPaid) {
+
+	// Takes branding from last possible card, in case earlier ones overriden.
+	let cardList = document.createDocumentFragment();
+
+	cardsInfo.articles.forEach((info, idx) => {
+		cardList.appendChild(buildCard(info, idx, isPaid));
+	});
+
+	// DOM mutation function.
+	return () => {
+
+		addBranding(cardsInfo.articles.slice(-1)[0], isPaid);
+		let advertRow = document.querySelector('.adverts__row');
+		advertRow.appendChild(cardList);
+		editionLink(cardsInfo.articles[0].edition, isPaid);
+
+	};
+
+}
+
+// Displays or hides a popup.
+function togglePopup (button, popup) {
+
+	if (button.classList.contains('is-active')) {
+
+		button.classList.remove('is-active');
+		popup.classList.add('is-off');
+
+	} else {
+
+		button.classList.add('is-active');
+		popup.classList.remove('is-off');
+
+	}
+
+}
+
+// Toggles the about information.
+function buttonListener () {
+
+	let aboutButton = document.querySelector('.popup__toggle');
+	let popup = document.querySelector('.popup');
+
+	aboutButton.addEventListener('click', () => {
+
+		write(() => {
+			togglePopup(aboutButton, popup)
+		});
+
+	});
+
+}
+
+export default function capiMultiple (adType) {
+
+	const isPaid = (adType === 'paidfor');
+
+	getIframeId()
+	.then(retrieveCapiData)
+	.then(capiData => buildCards(capiData, isPaid))
+	.then(write)
```

Comment:
I've just remembered why I included `linkEdition` here, it's because it needs access to the `edition` value from cAPI; the same is true for `addBranding`. I've renamed the function to make it clearer what it's doing.